src/test/java/com/edmunds/tools/databricks/maven/ImportWorkspaceMojoTest.java

```diff
@@ -56,18 +65,37 @@ public void testImportWorksWithProperPath() throws Exception {
     // Cannot use expected exceptions with a MojoExecutionException...
     @Test
     public void testImportFailsWithImproperPath() throws Exception {
+
+        ImportWorkspaceMojo underTest = getNoOverridesMojo(GOAL, "_fails_validation");
+
         File workspacePath = new File(this.getClass().getResource("/notebooks")
-            .getPath());
+                .getPath());
         underTest.validate = true;
         underTest.setSourceWorkspacePath(workspacePath);
-        Mockito.when(project.getGroupId()).thenReturn("com.edmunds.nottest");
-        Mockito.when(project.getArtifactId()).thenReturn("mycoolartifact");
-        underTest.setProject(project);
         try {
             underTest.execute();
         } catch (MojoExecutionException e) {
+            assertThat(e.getMessage(), containsString("JOB NAME VALIDATION FAILED [ILLEGAL VALUE]:\n" +
```

Comment:
ah I don't know if you tried doing the exceptions as part of the annotation... it definitely works with mojo exceptions and I am not sure what about the specific case I was testing that wasn't working but I had jumped to incorrect conclusions. Sorry for the confusion.